packages/main/src/plugin/container-registry.spec.ts

```diff
@@ -88,3 +124,32 @@ test('push should succeed if provider', async () => {
   const result = await containerRegistry.pushImage('dummy', 'image:latest', () => {});
   expect(result).toBeUndefined();
 });
+
+test('restartContainersByLabel should succeed successfully if project name is provided and call restartContainer', async () => {
+  const engine = {
+    // Fake that we have 3 containers of the same project
+    listSimpleContainers: vi
+      .fn()
+      .mockResolvedValue([
+        fakeContainerWithComposeProject,
+        fakeContainerWithComposeProject,
+        fakeContainerWithComposeProject,
+      ]),
+    getContainer: vi.fn().mockReturnValue({ restart: vi.fn().mockResolvedValue({}) }),
+    listPods: vi.fn().mockResolvedValue([]),
+    restartContainer: vi.fn().mockResolvedValue({}),
+  };
+  vi.spyOn(containerRegistry, 'getMatchingEngine').mockReturnValue(engine as unknown as Dockerode);
+  vi.spyOn(containerRegistry, 'listSimpleContainers').mockReturnValue(engine.listSimpleContainers());
+
+  // Spy on restartContainer to make sure it's called
+  // it is NOT called if there are no matches.. So it's important tot check this.
+  const restartContainer = vi.spyOn(containerRegistry, 'restartContainer');
+
+  // Restart all containers in the 'project1' project
+  const result = await containerRegistry.restartContainersByLabel('dummy', 'com.docker.compose.project', 'project1');
+  expect(result).toBeUndefined();
+
+  // Expect restartContainer tohave been called 3 times
+  expect(restartContainer).toHaveBeenCalledTimes(3);
```

Comment:
should it check that we have only 3 but also that we do have in listContainer result one container without the matching label ?